source/module_cell/klist.h

```diff
@@ -29,60 +102,121 @@ class K_Vectors
     K_Vectors();
     ~K_Vectors();
 
-    void set(
-        const ModuleSymmetry::Symmetry &symm,
-        const std::string &k_file_name,
-        const int& nspin,
-        const ModuleBase::Matrix3 &reciprocal_vec,
-        const ModuleBase::Matrix3 &latvec);
+    void set(const ModuleSymmetry::Symmetry &symm,
+             const std::string &k_file_name,
+             const int& nspin,
+             const ModuleBase::Matrix3 &reciprocal_vec,
+             const ModuleBase::Matrix3 &latvec);
 
-    void ibz_kpoint(const ModuleSymmetry::Symmetry &symm, bool use_symm,std::string& skpt, const UnitCell &ucell, bool& match);
+    void ibz_kpoint(const ModuleSymmetry::Symmetry &symm, 
+                    bool use_symm,std::string& skpt, 
+                    const UnitCell& ucell, 
+                    bool& match);
     //LiuXh add 20180515
-    void set_after_vc(
-            const ModuleSymmetry::Symmetry &symm,
-            const std::string &k_file_name,
-            const int& nspin,
-            const ModuleBase::Matrix3 &reciprocal_vec,
-            const ModuleBase::Matrix3 &latvec);
+    void set_after_vc(const ModuleSymmetry::Symmetry &symm,
+                     const std::string &k_file_name,
+                     const int& nspin,
+                     const ModuleBase::Matrix3 &reciprocal_vec,
+                     const ModuleBase::Matrix3 &latvec);
     //get global index for ik
     inline int getik_global(const int& ik) const;
 
-    
-    int get_nks() const
-    {
-        return this->nks;
-    }
+// REFACTORING K_Vectors functions below >>>
+private:
+    // newly built functions on refactor to remove GlobalC::Pkpoints and parallel_kpoints class
+    K_Vectors(const std::string& fkpt,
+              const double& lat0,
+              const std::vector<std::vector<double>>& avecs,
+              const std::vector<std::vector<double>>& bvecs,
+              const int& npools, 
+              const int& nprocs_inpool);
 
-    int get_nkstot() const
-    {
-        return this->nkstot;
-    }
+    // partial alternative to set() function
+    bool build_kpt(const std::string& fkpt);
 
-    int get_nkstot_ibz() const
-    {
-        return this->nkstot_ibz;
-    }
+    // I/O functions
+    // read ABACUS KPT file
+    void read_abacus_kpt(const std::string& fkpt);
+    // from special kpoints specified in KPT file, generate a kpoint-path
+    static void interpolate_knodes(const std::vector<std::vector<double>>& knodes, //< [in] special kpoints direct coordinates
```

Comment:
I am very confused of this code block, there are many static functions and non-static functions with private limits.
As I understand it, it should appear in the cpp file as a separate function but not .h file.

Reply:
Another question is that these functions are commented well but cannot be called by developers?

Reply:
> Another question is that these functions are commented well but cannot be called by developers?

Actually the full substitution of functions in klist has not completed yet, so they are all hidden with "private" declaration. Once substitution completes, old functions will be deleted. 
Seems I should substitute some functions for now and others in other times.